dumpling/tests/basic/run.sh

```diff
@@ -155,10 +155,40 @@ run_dumpling -B "test_db" -L ${DUMPLING_OUTPUT_DIR}/dumpling.log
 cnt=$(grep "IOTotalBytes=" ${DUMPLING_OUTPUT_DIR}/dumpling.log | grep -v "IOTotalBytes=0" | wc -l)
 [ "$cnt" -ge 1 ]
 
-export GO_FAILPOINTS=""
+echo "Test for failing to close meta/data file"
+export GO_FAILPOINTS="github.com/pingcap/tidb/dumpling/export/FailToCloseMetaFile=1*return"
+rm ${DUMPLING_OUTPUT_DIR}/dumpling.log
+set +e
+run_dumpling -B "test_db" -L ${DUMPLING_OUTPUT_DIR}/dumpling.log
+set -e
+cnt=$(grep -w "dump failed error stack info" ${DUMPLING_OUTPUT_DIR}/dumpling.log|wc -l)
+[ "$cnt" -ge 1 ]
+
+# dumpling retry will make it succeed
+export GO_FAILPOINTS="github.com/pingcap/tidb/dumpling/export/FailToCloseDataFile=1*return"
 export DUMPLING_TEST_PORT=4000
+run_sql "drop database if exists test_db;"
+run_sql "create database test_db;"
+run_sql "create table test_db.test_table (a int primary key);"
+run_sql "insert into test_db.test_table values (1),(2),(3),(4),(5),(6),(7),(8);"
+rm ${DUMPLING_OUTPUT_DIR}/dumpling.log
+set +e
+run_dumpling -B "test_db" -L ${DUMPLING_OUTPUT_DIR}/dumpling.log
+set -e
+cnt=$(grep -w "dump data successfully" ${DUMPLING_OUTPUT_DIR}/dumpling.log|wc -l)
+[ "$cnt" -ge 1 ]
```

Comment:
maybe check the output file has exactly 8 rows

Reply:
addressed in https://github.com/pingcap/tidb/pull/45374/commits/2191553a36cd875d4f3b0bb7677dd02e4d7aff99